lnrpc/routerrpc/router_server_test.go

```diff
@@ -0,0 +1,169 @@
+package routerrpc
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/queue"
+	"github.com/lightningnetwork/lnd/routing"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+)
+
+func makeStreamMock() *StreamMock {
+	return &StreamMock{
+		ctx:            context.Background(),
+		sentFromServer: make(chan *lnrpc.Payment, 10),
+	}
+}
+
+type StreamMock struct {
```

Comment:
We now use `testify/mock` in our tests, check `routing/mock_test.go`. I think we could export the mockers there.

Reply:
There's one mock for the control tower and one for the rpc stream. Should I move the one for the control tower to `routing/mock_test.go` and create a new `lnrpc/routerrpc/mock_test.go`?

Reply:
We already have `mockControlTower` in `routing/mock_test.go`. So I think you could export that, renaming it to `MockControlTower` and use it here. As for the rpc stream, yeah, leave it here.

Reply:
That `mockControlTower` is in the `routing` package. The test for the `router_server` I made is in the `routerrpc` package. I can't import a type from another package in the tests (see [here](https://stackoverflow.com/a/22277807/3883866)). 

So there's 3 options, I think.
1) Leave the code as it is now, with a new mock for this specific test.
2) Always compile `mock_test` into the `routing` package.
3) Create a seperate package for the mocks in `mock_test`

I obviously prefer option 1, because that's no work, lol.
What do you think?

Reply:
Oh yeah forgot about that. In that case I'd prefer option 1 for the moment, just to keep the PR concise. Once we start patching unit tests for RPCs, the mockers could live inside `lntest/mock`, which I believe was intended for such a job.